blueflood-core/src/main/java/com/rackspacecloud/blueflood/eventemitter/RollupEventEmitter.java

```diff
@@ -30,20 +30,51 @@
 import java.util.Arrays;
 import java.util.concurrent.*;
 
+/**
+ * @todo {@code RollupEventEmitter} should be modified so that it can be used
+ * as a drop-in replacement for {@link Emitter<RollupEvent>}.
+ *
+ * The {@code RollupEventEmitter} class violates the Liskov substitution
+ * principle, in that it doesn't behave the same as
+ * {@link Emitter<RollupEvent>}. Moreover, the differences in behavior depend
+ * upon particular details of the inputs.
+ *
+ * For example, if there is no listener registered for
+ * {@link RollupEventEmitter#ROLLUP_EVENT_NAME}, then *ALL* events are silently
+ * ignored. No other listeners will ever be notified of their events, even
+ * events of other names.
+ *
+ * Also, if any event doesn't have a {@link RollupEvent#rollup} of type
+ * {@link BasicRollup}, that event will be silently ignored. This is the case
+ * if the event's {@link RollupEvent#rollup} field is {@code null}, or if it's
+ * another rollup type, such as
+ * {@link com.rackspacecloud.blueflood.types.HistogramRollup}
+ *
+ * @todo Merge {@code RollupEventEmitter} and {@link Emitter<>} into a single
+ * class. {@link Emitter<>} doesn't appear to be used anywhere else.
+ *
+ * @todo Change the {@code eventPayload}/{@code args} parameter of
+ * {@link RollupEventEmitter#emit} from an array to a single item. There don't
+ * appear to be any parts of the codebase that use more than a single event
+ * argument.
+ */
```

Comment:
@izrik - would you also make a story to document this work.  We might forget about it if its not in our backlog.